File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum=0;
    void func(TreeNode root){
        if(root==null) return;

        func(root.right);
        sum+=root.val;
        root.val=sum;
        func(root.left);
    }
    public TreeNode bstToGst(TreeNode root) {
        func(root);

        return root;
    }
}

File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum=0;
    void func(TreeNode* root){
        if(root==NULL) return;

        func(root->right);
        sum+=root->val;
        root->val=sum;
        func(root->left);
    }
    TreeNode* bstToGst(TreeNode* root) {
        func(root);

        return root;
    }
};